pytorch_lightning/trainer/training_loop.py

```diff
@@ -298,6 +298,11 @@ def training_step(self, split_batch, batch_idx, opt_idx, hiddens):
             args = self.build_train_args(split_batch, batch_idx, opt_idx, hiddens)
             training_step_output = self.trainer.accelerator_backend.training_step(args)
             training_step_output = self.trainer.call_hook('training_step_end', training_step_output)
+            if training_step_output is None:
+                rank_zero_info(
+                    'training_step returned None. This is unusual, so please verify this was your intention'
+                )
```

Comment:
should we do this log only once with some kind of boolean state? I mean one common use case where these logs might be weird is if someone is using 2 optimizers with separate model params and want to call `loss.backward()` only once. This came up yesterday on the slack channel I guess is a pretty common one.

Reply:
IMO if we want to do that. It's just easier to have it as `warnings.warn`.

So these are the options:

1. `warnings.warn`. appears once per run
2. `log.warning`. appears every time
3. `log.info`. appears every time
4. `log.warning` with manual filtering
    4a. appears once per run
    4b. appears once per epoch
5. `log.info` with manual filtering
    5a. appears once per run
    5b. appears once per epoch

Should we have a vote or something? I prefer 1 personally.

@williamFalcon @Borda @ananthsub

Reply:
or maybe use `rank_zero_warn` if it can be a warning? will appears once per run I guess.

Reply:
rank_zero_warn is option 1.

Reply:
rank_zero_warn is fine to me. A pet peeve I have with the `rank_zero_*` logging functions is we lose the line number and file name from the original callsite, but that needs to be addressed separately